### PR TITLE
Adds toggle show/hide password

### PR DESCRIPTION
**Pull request by saketkumar95**
Fixes : https://github.com/zulip/zulip-mobile/issues/531
Here is the demo.

demo

### CODE REVIEW

**Comment by borisyankov** [comment]
I thought you intend to do it via the 'eye in input' pattern?

Or it can be just a show/hide text:
https://cdn.cloudfour.com/wp-content/uploads/2013/06/polar-hideshow.png

**Comment by saketkumar95** [comment]
@borisyankov ZulipSwitch was looking perfect here , so I thought to have this. 😄

**Comment by nashvail** [comment]
We can put an eye icon next to the switch or better put the icon inside the Input itself.

**Comment by saketkumar95** [comment]
@nashvail Cool. How about Show/Hide Text? I think it would be better?

**Comment by nashvail** [comment]
@saketkumar95 that's even better as long as it doesn't take up a lot of space on small devices

**Comment by saketkumar95** [comment]
@nashvail Let me try both. Will update with the best one.

**Comment by saketkumar95** [comment]
@borisyankov Just forgot to mention that I did the changes but didn't added the new demo. Checkout the new one.

h

**Comment by borisyankov** [comment]
I think the idea was to show the button inside the Input itself.

**Comment by saketkumar95** [comment]
There is a issue with that. Suppose the password is bit long so typing it would overlap the Show/Hide text. Any solution for that?

**Comment by borisyankov** [comment]
Set `paddingRight` to a higher value.

**Comment by saketkumar95** [comment]
@borisyankov Updated the branch.Sorry for the delay, Travelling right now. But I will try to overcome this breaks. :)

File: src/start/PasswordAuthView.js [hunk 1]
```diff
@@ -43,7 +45,9 @@ class PasswordAuthView extends React.Component {
       progress: false,
       email: props.email || config.defaultLoginEmail,
       password: config.defaultLoginPassword,
-      error: ''
+      error: '',
+      ishidden: true,
+      toggleText: 'Show',
```
Comment (borisyankov) [inline src/start/PasswordAuthView.js]: You don't have to use both of these. You can determine the text from the 'isHidden' value.

File: src/start/PasswordAuthView.js [hunk 2]
```diff
@@ -43,7 +45,9 @@ class PasswordAuthView extends React.Component {
       progress: false,
       email: props.email || config.defaultLoginEmail,
       password: config.defaultLoginPassword,
-      error: ''
+      error: '',
+      ishidden: true,
```
Comment (borisyankov) [inline src/start/PasswordAuthView.js]: Letter casing should be 'isHidden'. Also it should probably be more descriptive, like 'isPasswordHidden'.

File: src/start/PasswordAuthView.js [hunk 3]
```diff
@@ -95,14 +109,21 @@ class PasswordAuthView extends React.Component {
...
+          style={styles.passwordfield}
           placeholder="Password"
-          secureTextEntry
+          secureTextEntry={this.state.ishidden}
           value={password}
           onChangeText={newPassword => this.setState({ password: newPassword })}
           blurOnSubmit={false}
           onSubmitEditing={this.validateForm}
         />
+        <View style={styles.togglePassword}>
```
Comment (borisyankov) [inline src/start/PasswordAuthView.js]: A better approach would be to have a `PasswordInput` component that uses `Input` and it contains the Show/Hide toggle inside.

Comment (saketkumar95) [inline src/start/PasswordAuthView.js]: @borisyankov I don't think this is useful. PasswordInput is only used once here. And for this creating a new component is not a good idea i guess?

**Comment by saketkumar95** [comment]
@borisyankov Updated!

File: src/start/PasswordAuthView.js (hunk 3 above)
Comment (borisyankov) [inline src/start/PasswordAuthView.js]: It is to separate the functionality that is related only to passwords.

File: src/start/PasswordAuthView.js [hunk 4]
```diff
@@ -77,6 +85,18 @@ class PasswordAuthView extends React.Component {
     }
   };
 
+  handleToggle = () => {
+    const { isPasswordHidden } = this.state;
+
+    if (isPasswordHidden) {
+      this.setState({ isPasswordHidden: false });
+      this.setState({ toggleText: 'Hide' });
```
Comment (borisyankov) [inline src/start/PasswordAuthView.js]: You are still using the two states?

**Comment by zulipbot-test** [comment]
Hello @saketkumar95, a Zulip maintainer reviewed this pull request over a week ago, but you haven't updated your pull request since. Please take a look at the requested changes and update your pull request accordingly.

Thank you for your valuable contributions to Zulip!

**Comment by borisyankov** [comment]
Let's finish up this PR?

**Comment by zulipbot-test** [comment]
Hello @saketkumar95, a Zulip maintainer reviewed this pull request over a week ago, but you haven't updated your pull request since. Please take a look at the requested changes and update your pull request accordingly.

Thank you for your valuable contributions to Zulip!

**Comment by borisyankov** [comment]
This was implemented in a different way, so closing this.

**Pull request closed without merging**